### ISSUE TITLE
Guest Author Page

### ISSUE THREAD

**Issue by MikeNGarrett** [Automattic/Co-Authors-Plus#201]
/author/[author_slug] is no longer working for guest authors. The template is returning a 404.

**Comment by maxfenton** [comment]
Running into the same problem.

**Comment by maxfenton** [comment]
I haven't been able to test @matthewspencer's fix, so I basically reverted https://github.com/Automattic/Co-Authors-Plus/commit/2ce2ce88f75d848440da2395cd3bbe375bdac537 by making a custom function in my theme `functions.php` with a modified version of the older `fix_author_page`

```
remove_filter( 'posts_selection', array( $coauthors_plus, 'fix_author_page' ) );
remove_action( 'the_post', array( $coauthors_plus, 'fix_author_page' ) );
add_filter( 'posts_selection', 'mf_fix_author_page' ); 
add_action( 'the_post', 'mf_fix_author_page' );
function mf_fix_author_page() { /* ... */ }
```

**Comment by rebeccahum** [comment]
@MikeNGarrett  Are you still experiencing this issue? Have you tried @matthewspencer's fix?  That wp cli command should update the guest author's posts with the correct amount.

**Comment by MikeNGarrett** [comment]
This was so long ago that I'm not sure the context of the original issue. I know we found a way to work around the issue way back then, but I'm not sure if the original issue is fixed.

**Comment by jqueryalmeida** [comment]
Solution:
`function co_author_archive($query) {
    if ($query->is_author)
        $query->set( 'post_type', array('custom_post', 'post') );
    remove_action( 'pre_get_posts', 'co_author_archive' );
}
add_action('pre_get_posts', 'co_author_archive');
`

**Comment by rebeccahum** [comment]
Hi @jqueryalmeida, thanks so much for your contribution. Would you be able to submit a PR for this?

**Comment by trepmal** [comment]
I think this is resolved in #440.
